app/javascript/gobierto_data/webapp/components/sidebar/SidebarCategories.vue

```diff
@@ -79,22 +85,114 @@ export default {
         || (this.filters[1] && (this.filters[1] || {}).count >= 1));
     },
-    filtersModify() {
-      return this.filters.length ? this.filters.map(d => ({ ...d, isToggle: true })) : []
+    isAllChecked() {
+      return this.routeItemsFrequency.length !== 0 || this.routeItemsCategory.length !== 0
+    }
+  },
+  created() {
+    if (this.$route.params.pathMatch !== undefined) {
+      let paramsRouter = this.$route.params.pathMatch.split(':')
+      this.isPermalinkActive = true
+      this.selectedCheckbox(paramsRouter)
     }
   },
   methods: {
     //TODO temporary functions, waiting for the filter refactor
     sendCheckboxStatus_TEMP({ id, value, filter }) {
       this.$root.$emit("sendCheckbox_TEMP", { id, value, filter })
-      // eslint-disable-next-line no-unused-vars
-      this.$router.push('/datos').catch(err => {})
+
+      this.updateURLwithSelectedCategories(filter)
+      this.checkSelectedCheckbox()
     },
     selectAllCheckbox_TEMP({ filter }) {
       this.$root.$emit("selectAll_TEMP", { filter })
+
+      this.updateURLwithSelectedCategories(filter)
+      this.checkSelectedCheckbox()
+    },
+    updateURLwithSelectedCategories(values) {
+      if (this.$route.name === 'Dataset') {
+        // eslint-disable-next-line no-unused-vars
+        this.$router.push('/datos/').catch(err => {})
+      }
+
+      const { options: optionsChecked } = values
+      const { key } = values
+      //We filter by selected elements
+      const isItemSelected = optionsChecked.filter(({ isOptionChecked, counter }) => isOptionChecked === true && counter > 0)
+      //Now we get only the id of them
+      const getIdFromItems = [...new Set(isItemSelected.map(({ id }) => id))]
+      this.selectedCheckbox(getIdFromItems)
+
+      //Create an array which contains only the id from selected elements
+      let routeItems = []
+      if (key === 'frequency') {
+        this.routeItemsFrequency = this.getElements(getIdFromItems)
+        this.routeItemsCategory = this.reFilterElements('category')
+      }
+
+      if (key === 'category') {
+        this.routeItemsCategory = this.getElements(getIdFromItems)
+        this.routeItemsFrequency = this.reFilterElements('frequency')
+      }
+
+      //Create an array with all elements
+      routeItems = [...this.routeItemsFrequency, ...this.routeItemsCategory]
+      //Remove duplicate elements, and remove commas
+      routeItems = [...new Set(routeItems)];
+      routeItems = routeItems.toString().replace(/,/gi, '')
+
+      //If the user accesses through a permalink we change the route
+      let backSlashRoute = this.$route.fullPath === '/datos' ? `${this.$route.path}/terms/` : `${this.$route.path}terms/`
+      let urlTerms = this.isPermalinkActive ? `${location.origin}/datos/terms/` : backSlashRoute
+      urlTerms = routeItems.length === 0 ? `${this.$route.path}` : urlTerms
+
+      history.pushState(
+        {},
+        null,
+        `${urlTerms}${routeItems}`
+      )
```

Comment:
Is vue-router notified somehow? Mixing these things causes issues often

Reply:
Nope, to avoid errors i've replaced it.